onboarding/source/amb_light_service.c

```diff
@@ -17,37 +21,126 @@
+    if (lightServiceTaskHandle == NULL)
+    {
+        lightServiceTaskHandle = xTaskCreateStatic(lightServiceTask,
+                                                   LIGHT_SERVICE_NAME,
+                                                   LIGHT_SERVICE_STACK_SIZE,
+                                                   NULL,
+                                                   LIGHT_SERVICE_PRIORITY,
+                                                   lightServiceTaskStack,
+                                                   &lightServiceTaskBuffer);
+    }
+
+    if (lightServiceTaskHandle == NULL)
+    {
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    }
+
+    if (queueHandle == NULL)
+    {
+        queueHandle = xQueueCreateStatic(QUEUE_LENGTH,
+                                         ITEM_SIZE,
+                                         ucQueueStorageArea,
+                                         &queueBuffer);
+    }
 
+    if (queueHandle == NULL)
+    {
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
+    }
+
+    return OBC_ERR_CODE_SUCCESS;
     /* USER CODE END */
 }
 
-static void lightServiceTask(void * pvParameters) {
+uint32_t getLightSensorData(void)
+{
+    adcData_t adc_data;
+
+    /** - Start Group1 ADC Conversion
+     *     Select Channel 6 - Light Sensor for Conversion
+     */
+    adcInit();
+
+    /** - Wait for ADC Group1 conversion to complete */
+
+    adcStartConversion(adcREG1, adcGROUP1);
+    /** - Read the conversion result
+     *     The data contains the Light sensor data
+     */
+    while (!adcIsConversionComplete(adcREG1, adcGROUP1))
+        ;
+
+    /** - Transmit the Conversion data to PC using SCI
+     */
+    adcGetData(adcREG1, adcGROUP1, &adc_data);
+    return adc_data.value;
+}
+
+static void lightServiceTask(void *pvParameters)
+{
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    if (queueHandle != NULL)
+    {
+        light_event_t lightEvent;
+        while (1)
+        {
+            // queueBuffer has different type with event, will it convert automatically?
+            if (xQueueReceive(queueHandle, &lightEvent, (TickType_t)0) == pdTRUE)
+            {
+                if (lightEvent == MEASURE_LIGHT)
+                {
+                    uint32_t lightData = getLightSensorData();
+                    sciPrintf("%u\n", lightData);
+                };
+            }
+        }
+    }
 
     /* USER CODE END */
 }
 
-obc_error_code_t sendToLightServiceQueue(light_event_t *event) {
+obc_error_code_t sendToLightServiceQueue(light_event_t *event)
+{
     /* USER CODE BEGIN */
     // Send the event to the queue. Return error code if event was not sent successfully.
-    
+    if (queueHandle != NULL)
+    {
+        if (xQueueSend(queueHandle, event, (TickType_t)0) == pdTRUE)
+        {
+            // Are there any other issue that may cause queuesend fail?
```

Comment:
In this specific program, the queue will probably never get full. However, you should always check if you sent the event successfully (like you've done here). If the queue is full, then `xQueueSend` would return false. It's good that you returned `OBC_ERR_CODE_QUEUE_SEND_FAILED` in that case.